src/drizzleReactHooks.js

```diff
@@ -0,0 +1,203 @@
+import React, {
+  createContext,
+  useCallback,
+  useContext,
+  useEffect,
+  useMemo,
+  useRef,
+  useState
+} from 'react'
+import PropTypes from 'prop-types'
+import debounce from 'debounce'
+import shallowequal from 'shallowequal'
+
+const Context = createContext()
+export const useDrizzle = () => useContext(Context)
+export const useDrizzleState = (mapState, args) => {
+  const { drizzle } = useDrizzle()
+  const mapStateRef = useRef(mapState)
+  mapStateRef.current = mapState
+  const argsRef = useRef(args)
+  const [state, setState] = useState(
+    mapStateRef.current(drizzle.store.getState())
+  )
+  const stateRef = useRef(state)
+  if (!shallowequal(argsRef.current, args)) {
```

Comment:
This looks like an optimization of some sort, but I'm not sure how it works. Consider refactoring and adding some comments to describe what is happening.